metalwihen/WordPress-Android
# Introduce ViewModel for Prompts List screen

Fixes #17125 

This PR introduces the following changes:
- Refactor the UI approach to support visibility flags in viewmodel
- Introduce the ViewModel with tests
- Add a retry button in the Error and NoConnection error states
- Add some minor refactoring: (1) moving the classes to a different package. (2) Rename viewmodels to distinguish between ParentViewModel (to share info across activity, parent fragment and multiple instances of list fragments) and ViewModel (to handle presentation logic in list fragments)

**Note** - Please go through the commits one by one to make it easier to review. 

**To test:**
- Follow test instructions as explained in previous PRs: #2, #5, #6, #7, #8
- Ensure the UI matches the original requirements + one change (retry button shown for Error and NoConnection states)
- Tap on 'All' tab to view **No Connection** state
- Tap on Retry button on 'All' tab to view **Loaded** state
- Tap on 'Answered' tab to view **Loading** state
- Tap on `Not Answered` tab to view **Empty** state

**Screenshot:**
| Loading | Loaded | Error | Empty | No Connection |
| ----------- | ----------- | ----------- | ----------- | ----------- |
| ![image](https://user-images.githubusercontent.com/2140539/196770153-bac9dc03-a612-4300-9acb-59e35a49a997.png) | ![image](https://user-images.githubusercontent.com/2140539/196770304-824312ea-60e7-4d2f-a0fa-c3bc635a8973.png) | ![image](https://user-images.githubusercontent.com/2140539/196770431-0b28c4a7-8cfe-404d-86ed-cb9f8aa0d610.png) | ![image](https://user-images.githubusercontent.com/2140539/196770491-62adb38c-9b85-4314-a56c-0df055d6c066.png) | ![image](https://user-images.githubusercontent.com/2140539/196770551-1da916e1-ece1-4982-8e53-115b3a8b56f2.png) |

## Regression Notes
1. Potential unintended areas of impact
- PromptsListScreen flow should be retested since the classes were moved to another package. 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Manual testing around the Prompt List flow. 
- Run unit tests

3. What automated tests I added (or what prevented me from doing so)
- Unit tests were added to the ViewModel used in the PromptsListFragment

PR submission checklist:
